src/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -0,0 +1,412 @@
+﻿// ------------------------------------------------------------------------------
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.Identity.Client.Http;
+
+namespace Microsoft.Identity.Client.AppConfig
+{
+    /// <summary>
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public abstract class AbstractApplicationBuilder<T>
+        where T : AbstractApplicationBuilder<T>
+    {
+        internal AbstractApplicationBuilder(ApplicationConfiguration configuration)
+        {
+            Config = configuration;
+        }
+
+        internal ApplicationConfiguration Config { get; }
+
+        /// <summary>
+        /// </summary>
+        /// <param name="httpClientFactory"></param>
+        /// <returns></returns>
+        public T WithHttpClientFactory(IMsalHttpClientFactory httpClientFactory)
```

Comment:
See other comment around existing IHttpClientFactory interface in .net core.  we need to resolve if we're using the built-in one or defining our own.  if we define our own, it should have a unique name to avoid namespace collisions or having to use fully qualified names to resolve properly.

---
In reply to: [245862278](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/761#discussion_r245862278) [](ancestors = 245862278)